packages/markdown-slate/src/slateToCiceroMarkDom.js

```diff
@@ -19,18 +19,23 @@ const NS_CICERO = 'org.accordproject.ciceromark';
 
 
 /**
- * Removed the final node if it is an empty paragraph following a clause
+ * Removes nodes if they are an empty paragraph
  * @param {*} input the current result of slateToCiceroMarkDom
  * @returns {*} the final result of slateToCiceroMarkDom
  */
-const fixEmptyParaAfterClause = (input) => {
-    const nodes = input.nodes;
-    if (nodes.length >= 2 &&
-        nodes[nodes.length - 2].$class === 'org.accordproject.commonmark.Clause' &&
-        nodes[nodes.length - 1].$class === 'org.accordproject.commonmark.Paragraph' &&
-        nodes[nodes.length - 1].text === '') {
-        input.nodes = nodes.splice(0, nodes.length-1);
-    }
+const removeEmptyParagraphs = (input) => {
+    const emptyParagraph = {
+        '$class': 'org.accordproject.commonmark.Paragraph',
+        nodes: [
+            {
+                '$class': 'org.accordproject.commonmark.Text',
+                text: ''
+            }
+        ]
+    };
+
+    input.nodes = input.nodes.filter(node => node !== emptyParagraph);
```

Comment:
You cannot do equality like this in JavaScript. This will _not_ check that the node has the same structure as an empty paragraph, but whether the two nodes are the same object in memory.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators

Reply:
Ah yes! That was a weird mistake. I could `JSON.stringify` the two nodes and compare since they will be in the same order.

Reply:
How do we know they will be in the same order?

Reply:
Wouldn't the deleted code do the right thing?